src/BulkWriteCommandBuilder.php

```diff
@@ -0,0 +1,226 @@
+<?php
+
+namespace MongoDB;
+
+use MongoDB\Codec\DocumentCodec;
+use MongoDB\Codec\Encoder;
+use MongoDB\Driver\BulkWriteCommand;
+use MongoDB\Driver\Manager;
+use MongoDB\Exception\InvalidArgumentException;
+
+use function is_array;
+use function is_bool;
+use function is_string;
+
+final readonly class BulkWriteCommandBuilder
```

Comment:
To debate the name of this class. I find it long and a bit technical. How about `BulkWriteBuilder` or even shorter `MongoDB\BulkWrite`?

"command" and "builder" are concepts that don't need to be expressed.

This is important as it's the entrypoint of this feature.

Reply:
Happy to revisit this before we settle on an API.

I agree that a Builder suffix is superfluous. I'd be open to moving this within the `MongoDB\Builder` namespace if you don't think that'd be problematic with all of the other classes there.

I think having "ClientBulkWrite" or a "Command" suffix may be relevant, though, as I'm worried about the need to distinguish this from the `Collection::bulkWrite()` API. I suppose we have more flexibility here, but I felt compelled to do so in PHPC (it doesn't sit well with me, but I didn't see an alternative).